routes/transaction.go

```diff
@@ -626,6 +777,238 @@ func GetBalanceForPublicKeyUsingUtxoView(
+		if err != nil || len(profilePublicKeyBytes) != btcec.PubKeyBytesLenCompressed {
+			_AddBadRequestError(ww, fmt.Sprintf(
+				"SubsidizedUpdateProfile: Problem decoding profile public key %s: %v",
+				requestData.ProfilePublicKeyBase58Check, err))
+			return
+		}
+	}
+	if requestData.UpdaterPublicKeyBase58Check != "" {
+		updaterPublicKeyBytes, _, err = lib.Base58CheckDecode(requestData.UpdaterPublicKeyBase58Check)
+		if err != nil || len(updaterPublicKeyBytes) != btcec.PubKeyBytesLenCompressed {
+			_AddBadRequestError(ww, fmt.Sprintf(
+				"SubsidizedUpdateProfile: Problem decoding updater public key %s: %v",
+				requestData.UpdaterPublicKeyBase58Check, err))
+			return
+		}
+	}
+
+	// Grab the profile being updated public key.
+	profilePublicKey := updaterPublicKeyBytes
+	if requestData.ProfilePublicKeyBase58Check != "" {
+		profilePublicKey = profilePublicKeyBytes
+	}
+
+	// NOTE: We cannot simply copy and paste the existing logic from UpdateProfile
+	// as some of that logic is dependent on the user's balance being non-zero. Since
+	// we use this endpoint to subsidize the user's update profile, we instead speculatively
+	// construct the update profile transaction for the user.
+
+	// (1) Validate and construct the update profile transaction.
+	var updateProfileTxn *lib.MsgDeSoTxn
+
+	// (1a) Validate the request data.
+	if err := fes.ValidateAndConvertUpdateProfileRequest(&requestData, profilePublicKey, utxoView); err != nil {
+		_AddBadRequestError(ww,
+			fmt.Sprintf("SubsidizedUpdateProfile: Problem validating request data: %v", err))
+		return
+	}
+
+	// (1b) Compute the transaction fees for this update profile transaction and encode the associated extra data.
+	additionalOutputs, err :=
+		fes.getTransactionFee(lib.TxnTypeUpdateProfile, updaterPublicKeyBytes, requestData.TransactionFees)
+	if err != nil {
+		_AddBadRequestError(ww,
+			fmt.Sprintf("SubsidizedUpdateProfile: TransactionFees "+
+				"specified in Request body are invalid: %v", err))
+		return
+	}
+	extraData, err := EncodeExtraDataMap(requestData.ExtraData)
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("SubsidizedUpdateProfile: Problem encoding ExtraData: %v", err))
+		return
+	}
+
+	// (1c) Determine if a profile creation fee must be paid.
+	additionalFees := uint64(0)
+	if existingProfileEntry := utxoView.GetProfileEntryForPublicKey(profilePublicKey); existingProfileEntry == nil {
+		additionalFees = utxoView.GetCurrentGlobalParamsEntry().CreateProfileFeeNanos
+	}
+
+	// (1c) Construct the update profile transaction.
+	updateProfileTxn, _, _, _, err = fes.blockchain.CreateUpdateProfileTxn(
+		updaterPublicKeyBytes,
+		profilePublicKeyBytes,
+		requestData.NewUsername,
+		requestData.NewDescription,
+		requestData.NewProfilePic,
+		requestData.NewCreatorBasisPoints,
+		requestData.NewStakeMultipleBasisPoints,
+		requestData.IsHidden,
+		additionalFees,
+		extraData,
+		requestData.MinFeeRateNanosPerKB,
```

Comment:
We should force this txn to be constructed with the global min fee rate, otherwise the subsidizing wallet could be drained very quickly.

Reply:
+1